src/TicketBai/Mews.Fiscalizations.TicketBai/TicketBaiClient.cs

```diff
@@ -0,0 +1,163 @@
+            (byte) 0xC1, (byte) 0xC6, (byte) 0xCF, (byte) 0xC8, (byte) 0xDD, (byte) 0xDA, (byte) 0xD3, (byte) 0xD4,
+            (byte) 0x69, (byte) 0x6E, (byte) 0x67, (byte) 0x60, (byte) 0x75, (byte) 0x72, (byte) 0x7B, (byte) 0x7C,
+            (byte) 0x51, (byte) 0x56, (byte) 0x5F, (byte) 0x58, (byte) 0x4D, (byte) 0x4A, (byte) 0x43, (byte) 0x44,
+            (byte) 0x19, (byte) 0x1E, (byte) 0x17, (byte) 0x10, (byte) 0x05, (byte) 0x02, (byte) 0x0B, (byte) 0x0C,
+            (byte) 0x21, (byte) 0x26, (byte) 0x2F, (byte) 0x28, (byte) 0x3D, (byte) 0x3A, (byte) 0x33, (byte) 0x34,
+            (byte) 0x4E, (byte) 0x49, (byte) 0x40, (byte) 0x47, (byte) 0x52, (byte) 0x55, (byte) 0x5C, (byte) 0x5B,
+            (byte) 0x76, (byte) 0x71, (byte) 0x78, (byte) 0x7F, (byte) 0x6A, (byte) 0x6D, (byte) 0x64, (byte) 0x63,
+            (byte) 0x3E, (byte) 0x39, (byte) 0x30, (byte) 0x37, (byte) 0x22, (byte) 0x25, (byte) 0x2C, (byte) 0x2B,
+            (byte) 0x06, (byte) 0x01, (byte) 0x08, (byte) 0x0F, (byte) 0x1A, (byte) 0x1D, (byte) 0x14, (byte) 0x13,
+            (byte) 0xAE, (byte) 0xA9, (byte) 0xA0, (byte) 0xA7, (byte) 0xB2, (byte) 0xB5, (byte) 0xBC, (byte) 0xBB,
+            (byte) 0x96, (byte) 0x91, (byte) 0x98, (byte) 0x9F, (byte) 0x8A, (byte) 0x8D, (byte) 0x84, (byte) 0x83,
+            (byte) 0xDE, (byte) 0xD9, (byte) 0xD0, (byte) 0xD7, (byte) 0xC2, (byte) 0xC5, (byte) 0xCC, (byte) 0xCB,
+            (byte) 0xE6, (byte) 0xE1, (byte) 0xE8, (byte) 0xEF, (byte) 0xFA, (byte) 0xFD, (byte) 0xF4, (byte) 0xF3
+        };
+
+        public TicketBaiClient(X509Certificate2 certificate, Environment environment)
+        {
+            Environment = environment;
+            SendInvoiceUri = environment.Match(
+                Environment.Production, _ => "https://tbai-z.egoitza.gipuzkoa.eus/sarrerak/alta",
+                Environment.Test, _ => "https://tbai-z.prep.gipuzkoa.eus/sarrerak/alta/"
+            );
+            Certificate = certificate;
+
+            var requestHandler = new HttpClientHandler();
+            requestHandler.ClientCertificates.Add(certificate);
+            HttpClient = new HttpClient(requestHandler);
+        }
+
+        private string SendInvoiceUri { get; }
+
+        private HttpClient HttpClient { get; }
+
+        private X509Certificate2 Certificate { get; }
+
+        private Environment Environment { get; }
+
+        // TODO: Return ITry and handle error responses?.
+        public async Task<SendInvoiceResponse> SendInvoiceAsync(SendInvoiceRequest request)
+        {
+            var ticketBaiRequest = ModelToDtoConverter.Convert(request);
+            var xmlDoc = new XmlDocument();
+
+            using (var memStm = new MemoryStream())
+            {
+                var serializer = new XmlSerializer(typeof(Dto.TicketBai));
+                serializer.Serialize(memStm, ticketBaiRequest);
+
+                memStm.Position = 0;
+
+                using (var xtr = new StreamReader(memStm, Encoding.UTF8))
```

Comment:
This stuff looks a bit weird. We have to manually set the position of the stream to 0 in order to read it? Is that how other fiscalizations read requests too?